### DISCUSSION TITLE
get status of multiple jobs in the same workflow

### DISCUSSION THREAD

**Discussion by carbohydrates** [community/community#25970] [Actions]
<p>Hello folks,</p>
<p>I have the workflow that contains 3 jobs, and I want to have 4th job that will send a slack message for me.</p>
<p>Could you please advise how to get the status of all jobs and their statuses that currently is on workflow.</p>
<p>The suggestion about putting all the jobs in one not an option</p>
<pre><code>name: CI
on:
  push:
    branches:
      - master
jobs:
  one:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v1
      - run: echo "one"
  two:
    runs-on: ubuntu-latest
    steps:
      - run: echo "two"
  five:
    runs-on: ubuntu-latest
    needs: [one, two]
    if: always()
    steps:
      - run: echo "How to know the status of execution of jobs.one and jobs.two?"
</code></pre>

**Comment by carbohydrates** [comment]
<p>The one thing that I found is using of Check API <a href="https://developer.github.com/v3/activity/events/types/#checksuiteevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#checksuiteevent</a></p>
<p>but this approach looks like a dirty workaround</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Now, github actions provide workflow jobs APIs , you could use it to list all the jobs of a workflow run. </p>
<p><a href="https://developer.github.com/v3/actions/workflow_jobs/#list-jobs-for-a-workflow-run" rel="nofollow noopener">https://developer.github.com/v3/actions/workflow_jobs/#list-jobs-for-a-workflow-run</a></p>
<p>And there is an environment variable <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">GITHUB_RUN_ID</a> you could use to get the current workflow run id. </p>
<p>You could get the conclusion element value for job status. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/4/4719dda73752811691a54694f7927e9468119288.png" data-download-href="/uploads/short-url/a8ZjHd8weej2f9S4w9QGFlKW0IM.png?dl=1" title="job conclusion.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181094349-14cfb325-dae3-4820-bf32-a1fe7ea23f0a.png" alt="job conclusion.png" data-base62-sha1="a8ZjHd8weej2f9S4w9QGFlKW0IM" width="567" height="499" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/4/4719dda73752811691a54694f7927e9468119288_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">job conclusion.png</span><span class="informations">779×686 34.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>There is another method to pass job status through artifacts, for more information, please check BrightRan’s solution here: </p>
<p><a href="https://github.community/t/how-get-the-status-of-previous-jobs/17555/2">https://github.community/t5/GitHub-Actions/How-get-the-status-of-previous-jobs/m-p/45511#M6237</a> </p>

**Comment by jenschelkopf** [comment]
<p>Take a look at the Workflow Conclusion Action. It checks the status of the other jobs in your workflows, and will create an environment variable with a rolled up status. It works well if you need to do something conditionally on the basis of the other jobs.</p>
<p><a href="https://github.com/marketplace/actions/workflow-conclusion-action" rel="nofollow noopener">https://github.com/marketplace/actions/workflow-conclusion-action</a></p>

**Comment by outrunthewolf** [comment]
<p>Did you find a reasonable solution to this?</p>

**Comment by adriaan-pelzer** [comment]
<p>Here’s my solution:</p>
<pre><code class="lang-auto">
jobs:

&amp;nbsp; job0:

&amp;nbsp; &amp;nbsp; name: job0

&amp;nbsp; &amp;nbsp; outputs:

&amp;nbsp; &amp;nbsp; &amp;nbsp; job-status: ${{ job.status }}

&amp;nbsp; job1:

&amp;nbsp; &amp;nbsp; name: job1

&amp;nbsp; &amp;nbsp; outputs:

&amp;nbsp; &amp;nbsp; &amp;nbsp; job-status: ${{ job.status }}

&amp;nbsp; job2:

&amp;nbsp; &amp;nbsp; name: job1

&amp;nbsp; &amp;nbsp; needs:

&amp;nbsp; &amp;nbsp; &amp;nbsp; - job0

&amp;nbsp; &amp;nbsp; &amp;nbsp; - job1

&amp;nbsp; &amp;nbsp; steps:

&amp;nbsp; &amp;nbsp; &amp;nbsp; - name: check job statuses

&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;run: |

&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;if [["${{ needs.job0.outputs.job-status }}" == "Success" &amp;&amp; "${{ needs.job1.outputs.job.status }} == "Success"]]; then

&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp;echo "Both jobs passed!"

&amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; &amp;nbsp; fi

</code></pre>

**Comment by femrtnz** [comment]
<p>Using the <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables" rel="nofollow noopener">GITHUB_RUN_ID</a> looks a good solution, however this is the run id and not the job id. What if have two jobs, how can we can get the right conclusion assuming the RUN_ID is the same?</p>

**Reply by yuval-katzman-digsec** [reply]
If anyone encounters this thread, this is an easier way to conditionally trigger a job based on the dependent jobs results (job.status):
`${{ contains(join(needs.*.result, ','), 'failure') }}`
this will evaluate to true if any of the dependent jobs has a result of "failure". this can be adapted further by changing the comparison to 'success' or by adding a '!' to the condition.

**Reply by aknosis** [reply]
Even simpler, you can remove the join:

`${{ contains(needs.*.result, 'failure') }}`, pulled from [docs](https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter)

**Reply by ghost** [reply]
> Daha da basiti, birleştirmeyi kaldırabilirsiniz:
> 
> `${{ contains(needs.*.result, 'failure') }}`[, dokümanlardan](https://docs.github.com/en/actions/learn-github-actions/expressions#example-using-an-object-filter) alınmıştır

**Reply by SimonHeimberg** [reply]
I like it. Just be aware of https://github.com/actions/runner/issues/1540:
`contains(needs.*.result, 'failure')` considers dependencies of dependencies (and the whole dependency chain) if it is in `if:` of a job. (But not if it is in `run:` or other places)

**Comment by RunningLeon** [comment]
Hi, does anyone know how to get the latest run status of all jobs in a same workflow and display them in the repo's `readme.md` as badages?

**Reply by rajat-mehraa** [reply]
If your use case is just that, you can use github badge api where they provide the latest workflow run status in svg format. 
https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

**Comment by rajat-mehraa** [comment]
I am looking for a way to get the status of a GHA along with the input parameters i passed during workflow dispatch. I am not sure how we can fetch the inputs

**Comment by mvgiacomello** [comment]
For my future self, this reports to Slack the results from any job (here `jobs-1` and `jobs-2`):
```
  slack:
    if: ${{ always() }}
    runs-on: ubuntu-latest
    needs: [jobs-1, jobs-2]
    steps:
      - name: Notify Slack
        uses: 8398a7/action-slack@v3
        with:
          status: custom
          fields: all
          custom_payload: |
            {
              attachments: [{
                color: "${{ contains(needs.*.result, 'failure') && '#A30201' || '#2eb886' }}",
                blocks: [
                  {
                    type: "section",
                    text: {
                      type: "mrkdwn",
                      text:
                      "${{ contains(needs.*.result, 'failure') }}"
                      ?
                      `Failure: (${process.env.AS_COMMIT}). \n Check the job: ${process.env.AS_JOB} for more info;`
                      :
                      `Success: (${process.env.AS_COMMIT}). \n by ${process.env.AS_AUTHOR}`
                    }
                  }
                ]
              }]
            }
```

**Comment by AThakur812001** [comment]
I am trying to setup a this github action having 4 jobs where 1st and 2nd and 3rd runs on basis of input during workflow_dispatch and last one should run if either of job1/job2/job3 is succeeded or skipped but on the basis of logic the last one is always getting skipped
Can someone help me here?

`name: Testing
run-name: "ONLY-TESTING"

on:
  workflow_dispatch:
    inputs:
      job_name:
        type: choice
        default: ''  
        options:
          - job1
          - job2
          - job3


jobs:
  job1:
    runs-on: tfg-builder
    if: ${{ inputs.job_name == 'job1' }}
    outputs:
      build_id: ${{ steps.build_step.outputs.build_id }}
    steps:
      - name: Build
        id: build_step
        run: echo "build_id=${{ github.run_number }}" >> $GITHUB_OUTPUT

  job2:
    if: ${{ inputs.job_name == 'job2' }}
    runs-on: tfg-builder
    steps:
      - run: echo "Deploying build ${{ needs.build.outputs.build_id }}"

  job3:
    if: ${{ inputs.job_name == 'job3' }}
    runs-on: tfg-builder
    steps:
      - run: echo "Hello WORLD from generic job3"

  fallback:
    needs: [ job1, job2, job3 ]
    runs-on: tfg-builder
    if: ${{ contains(join(needs.*.result, ','), 'success') }}
    # if: ${{ contains(needs.*.result, 'success') || contains(needs.*.result, 'skipped') }}
    steps:
      - run: echo "Previous jobs might have succeeded or skipped while build and deploy"`
